osu.Game/Online/OnlineViewContainer.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online.API;
+using osu.Game.Online.Placeholders;
+
+namespace osu.Game.Online
+{
+    /// <summary>
+    /// A <see cref="Container"/> for dislaying online content who require a local user to be logged in.
+    /// Shows its children only when the local user is logged in and supports displaying a placeholder if not.
+    /// </summary>
+    public class OnlineViewContainer : Container, IOnlineComponent
+    {
+        private readonly Container placeholderContainer;
+        private readonly Placeholder placeholder;
+        protected readonly LoadingAnimation LoadingAnimation;
+
+        private const int transform_time = 300;
+
+        protected Container View { get; }
```

Comment:
combine with content.

`protected override Container<Drawable> Content { get; }`

Reply:
Reverting suggested behaviour as it causes the container to not exhibit desired behaviour (Content doesn't hide) when `Content` property is overriden in sub-class

Reply:
i don't understand.

Reply:
Actually, nevermind it can be merged back. 
Care just needs to be taken to call `base.Content` when customizing the visibility animations of a drawable inheriting `OnlineViewContainer` and also overriding `Content` property. Else it would
cause the online view to still be displayed even if it should be hidden (as seen with a refactor of `Leaderboard<>` class inheriting `OnlineViewContainer`)

![image](https://user-images.githubusercontent.com/20256717/73608771-161c8e00-45c7-11ea-9b22-2d7bed003ea6.png)

Reply:
The component should be made in a way that this is not possible. Also, the change proposed here doesn't affect this – it could still happen with your current code.